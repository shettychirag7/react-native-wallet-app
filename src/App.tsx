import React from 'react';
import {SafeAreaView} from 'react-native';
import {Provider} from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

import {createStore} from './redux/store';

import MainStackNavigation from './navigation/MainStackNavigation';

import styles from './styles';

const App = () => {
  const store = createStore({
    enableMocks: true,
  });

  return (
    <SafeAreaView style={styles.container}>
      <Provider store={store}>
        <NavigationContainer>
          <MainStackNavigation />
        </NavigationContainer>
      </Provider>
    </SafeAreaView>
  );
};

export default App;
