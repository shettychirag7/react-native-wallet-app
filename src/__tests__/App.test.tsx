import React from 'react';
import {render} from '@testing-library/react-native';

import App from '../App';

/**
 * A high level test that ensures App renders without errors
 * Component and code level detailed unit tests can be found under their
 * respectively directory
 */
it('should render app with errors and match initial snapshot', () => {
  const {toJSON} = render(<App />);

  expect(toJSON()).toMatchSnapshot();
});
