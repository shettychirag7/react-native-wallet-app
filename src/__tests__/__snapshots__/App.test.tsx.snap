// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app with errors and match initial snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={true}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Main"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "white",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <RNSScreenNavigationContainer
                style={
                  {
                    "flex": 1,
                    "overflow": "hidden",
                  }
                }
              >
                <RNSScreen
                  activityState={2}
                  collapsable={false}
                  forwardedRef={[Function]}
                  gestureResponseDistance={
                    {
                      "bottom": -1,
                      "end": -1,
                      "start": -1,
                      "top": -1,
                    }
                  }
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "zIndex": 0,
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={false}
                    importantForAccessibility="auto"
                    style={
                      [
                        {
                          "backgroundColor": "white",
                          "flex": 1,
                        },
                        [
                          {
                            "flex": 1,
                            "flexDirection": "column-reverse",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            {
                              "fontSize": 16,
                              "lineHeight": 24,
                              "marginVertical": 8,
                            }
                          }
                        >
                          Home screen
                        </Text>
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "fontSize": 16,
                                  "lineHeight": 24,
                                  "marginVertical": 8,
                                },
                                {
                                  "color": "#2b6cd8",
                                },
                              ]
                            }
                          >
                            Go to notifications
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RNSScreen>
              </RNSScreenNavigationContainer>
              <View
                collapsable={false}
                onLayout={[Function]}
                pointerEvents="auto"
                style={
                  {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderTopColor": "rgb(216, 216, 216)",
                    "borderTopLeftRadius": 15,
                    "borderTopRightRadius": 15,
                    "borderTopWidth": 0.5,
                    "bottom": 0,
                    "elevation": 8,
                    "height": 75,
                    "left": 0,
                    "paddingBottom": 8,
                    "paddingHorizontal": 0,
                    "paddingTop": 10,
                    "position": null,
                    "right": 0,
                    "transform": [
                      {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  accessibilityRole="tablist"
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessibilityLabel="Home, tab, 1 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": true,
                      }
                    }
                    accessibilityStates={
                      [
                        "selected",
                      ]
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "#3aa1ee",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_home"
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "gray",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_home"
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#3aa1ee",
                          },
                          {
                            "fontSize": 10,
                          },
                          {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Home
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="Buy, tab, 2 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "#3aa1ee",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_storefront"
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "gray",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_storefront"
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "gray",
                          },
                          {
                            "fontSize": 10,
                          },
                          {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Buy
                    </Text>
                  </View>
                  <View
                    accessibilityLabel=", tab, 3 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <SvgMock />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <SvgMock />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "gray",
                          },
                          {
                            "fontSize": 10,
                          },
                          {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    accessibilityLabel="Transactions, tab, 4 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "#3aa1ee",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_payments"
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "gray",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_payments"
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "gray",
                          },
                          {
                            "fontSize": 10,
                          },
                          {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Transactions
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="Profile, tab, 5 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "#3aa1ee",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_person"
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "gray",
                                "fontSize": 24,
                              },
                              undefined,
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="icon_person"
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "gray",
                          },
                          {
                            "fontSize": 10,
                          },
                          {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Profile
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
